atomicfu/src/nativeMain/kotlin/kotlinx/atomicfu/locks/Synchronized.kt

```diff
@@ -129,12 +149,25 @@ public actual open class SynchronizedObject {
         init { freeze() }
     }
 
+    protected enum class Status { UNLOCKED, THIN, FAT }
+
     private fun CPointer<mutex_node_t>.lock() = lock(this.pointed.mutex)
 
     private fun CPointer<mutex_node_t>.unlock() = unlock(this.pointed.mutex)
 }
 
-private enum class Status { UNLOCKED, THIN, FAT }
+public actual fun reentrantLock() = ReentrantLock()
+
+public actual class ReentrantLock : SynchronizedObject()
```

Comment:
You don't need a separate class on Kotlin/Native. Use `typealias ReentrantLock = SynchronizedObject`.